src/Integration.UnitTests/MefServices/ActiveSolutionBoundTrackerTests.cs

```diff
@@ -499,31 +497,39 @@ public void SolutionBindingUpdated_WhenClearBoundProject_NotRaised()
             // Arrange
             using (var testSubject = new ActiveSolutionBoundTracker(this.host, this.activeSolutionTracker, loggerMock.Object))
             {
-                int callCount = 0;
-                testSubject.SolutionBindingUpdated += (sender, e) => callCount++;
+                var eventCounter = new EventCounter(testSubject);
 
                 // Act
                 host.VisualStateManager.ClearBoundProject();
 
                 // Assert
-                callCount.Should().Be(0);
+                eventCounter.PreSolutionBindingUpdatedCount.Should().Be(0);
+                eventCounter.SolutionBindingUpdatedCount.Should().Be(0);
+                eventCounter.PreSolutionBindingChangedCount.Should().Be(0);
+                eventCounter.SolutionBindingChangedCount.Should().Be(0);
             }
         }
 
         [TestMethod]
-        public void SolutionBindingUpdated_WhenSetBoundProject_Raised()
+        public void SolutionBindingUpdated_WhenSetBoundProject_EventsRaisedInExpectedOrder()
         {
             // Arrange
             using (var testSubject = new ActiveSolutionBoundTracker(this.host, this.activeSolutionTracker, loggerMock.Object))
             {
-                int callCount = 0;
-                testSubject.SolutionBindingUpdated += (sender, e) => callCount++;
+                var eventCounter = new EventCounter(testSubject);
 
                 // Act
                 host.VisualStateManager.SetBoundProject(new Uri("http://localhost"), null, "project123");
 
                 // Assert
-                callCount.Should().Be(1);
+                eventCounter.PreSolutionBindingUpdatedCount.Should().Be(1);
+                eventCounter.SolutionBindingUpdatedCount.Should().Be(1);
+                eventCounter.PreSolutionBindingChangedCount.Should().Be(0);
+                eventCounter.SolutionBindingChangedCount.Should().Be(0);
+
+                eventCounter.RaisedEventNames.Should().HaveCount(2);
+                eventCounter.RaisedEventNames[0].Should().Be("PreSolutionBindingUpdated");
+                eventCounter.RaisedEventNames[1].Should().Be("SolutionBindingUpdated");
```

Comment:
Additional check that the events are raised in the correct order.

Reply:
I'm not sure if this is definitive proof that it'll always work this way, but I can't think of a better way to do it, and this is better than nothing.

Reply:
The events are raised and handled synchronously on a single thread so the order should be predictable, shouldn't it? Or is there something that makes it non-deterministic?